app/gui/src/controller/searcher.rs

```diff
@@ -903,6 +954,19 @@ impl Searcher {
         executor::global::spawn(self.notifier.publish(Notification::NewActionList));
     }
 
+    fn init_input(&self) {
+        if let Mode::NewNode { source_node, .. } = self.mode.deref() {
+            if source_node.is_none() {
+                if let Err(e) = self.set_input("".to_string()) {
+                    tracing::error!(
+                        "Failed to clear input when creating searcher for a new node: {:?}",
+                        e
+                    );
+                }
+            }
+        }
+    }
```

Comment:
Why do we need this? Does the view make some input update before being cleared? IMO, whatever reason, it should be handled rather by presenter (to not propagate view input to searcher or AST changes to view).

Reply:
The input would otherwise be set to whatever is sued as the default value of a new node. I added a comment and moved this to the presenter.